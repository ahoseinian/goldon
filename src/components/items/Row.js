import React from 'react'
import {connect} from 'react-redux'
import {removeFromCart} from '../../actions'
import SmallImage from './image/SmallImage'
import Price from './Price'
import styled from 'styled-components'
import {color} from '../theme'
import Icon from '../common/Icon'

const Td = styled.td `
  vertical-align: middle !important;
`
const LeftBorder = styled.div `
  border-left: 1px solid ${color.light};
  margin-left: 1rem;
  padding-left: 1rem;
`

export const Row = ({id, images, name, quantity, price, removeFromCart}) => (
  <tr>
    <Td>
      { images &&
        <SmallImage id={images[0]}/>
      }
    </Td>
    <Td>{name}</Td>
    <Td>
      <div className="d-flex">
        <LeftBorder>
          {quantity}
        </LeftBorder>
        <Price value={price}/>
      </div>
      <Price value={price * quantity}/>
    </Td>
    <Td className="text-center remove-from-cart" onClick={() => removeFromCart(id)}>
      <Icon name="times" />
    </Td>
  </tr>
)

Row.propTypes = {
  id: React.PropTypes.string.isRequired,
  images: React.PropTypes.array,
  price: React.PropTypes.number.isRequired,
  quantity: React.PropTypes.number.isRequired,
  removeFromCart: React.PropTypes.func
}

export default connect(null, {removeFromCart})(Row)
