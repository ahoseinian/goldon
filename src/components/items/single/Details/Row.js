import React from 'react';

const SingleRow = ({name, value, measure}) => (
  <tr>
    <th>{name}</th>
    <td>{value} {measure}</td>
  </tr>
);

SingleRow.propTypes = {
  name: React.PropTypes.string.isRequired,
  value: React.PropTypes.oneOfType([
    React.PropTypes.number,
    React.PropTypes.string
  ]),
  measure: React.PropTypes.string
};

export default SingleRow;
