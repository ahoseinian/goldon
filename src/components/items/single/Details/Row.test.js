import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import Row from './Row'

describe('Row', () => {
  it('renders null without value', () => {
    const wrapper = shallow(<Row name="a" />)
    expect(wrapper.html()).toEqual(null)
  })

  it('renders correctly with value', () => {
    const wrapper = shallow(<Row name="a" value="test"/>)
    expect(wrapper.html().length > 0).toEqual(true)
    expect(toJson(
      wrapper
    )).toMatchSnapshot()
  })

})
