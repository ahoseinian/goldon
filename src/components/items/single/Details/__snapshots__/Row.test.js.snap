exports[`Row renders correctly with value 1`] = `
<tr>
  <th
    width="120">
    a
  </th>
  <td>
    test
     
  </td>
</tr>
`;
